pkg/scheduler/api/node_info.go

```diff
@@ -345,21 +346,28 @@ func (ni *NodeInfo) SetNode(node *v1.Node) {
 
 // setNodeOthersResource initialize sharable devices
 func (ni *NodeInfo) setNodeOthersResource(node *v1.Node) {
+	if node == nil {
+		return
+	}
+
 	ni.Others[GPUSharingDevice] = gpushare.NewGPUDevices(ni.Name, node)
 	ni.Others[vgpu.DeviceName] = vgpu.NewGPUDevices(ni.Name, node)
-	IgnoredDevicesList.Set(
-		ni.Others[GPUSharingDevice].(Devices).GetIgnoredDevices(),
-		ni.Others[vgpu.DeviceName].(Devices).GetIgnoredDevices(),
-	)
+
+	for _, val := range ni.Others {
+		if dev, ok := val.(Devices); ok {
+			IgnoredDevicesList.Set(dev.GetIgnoredDevices())
+		}
+	}
 }
 
 // setNode sets kubernetes node object to nodeInfo object without assertion
 func (ni *NodeInfo) setNode(node *v1.Node) {
+	ni.Name = node.Name
```

Comment:
Why adjust the location?

Reply:
Line 353 uses this parameter, I have noticed the situation of empty strings while observing the logs

Reply:
A little wired , `NewNodeInfo` has already set node name: )